### PR TITLE
[fix](func)fix array_with_const with larger than max_array_size

### PR DESCRIPTION
**Pull request by amorynan**
## Proposed changes
when we use func array_with_const with large size there has not limit so, we will make be core like this
截屏2024-07-08 20 41 43

Issue Number: close #xxx

### CODE REVIEW

**Comment by amorynan** [comment]
run buildall

File: regression-test/suites/datatype_p0/nested_types/query/array_functions/test_array_functions_array_with_const.groovy [hunk 1]
```diff
@@ -16,6 +16,7 @@
 // under the License.
 
 suite("test_array_functions_array_with_const", "p0") {
+    sql "set enable_nereids_planner=false;"
```
Comment (eldenmoon) [inline regression-test/suites/datatype_p0/nested_types/query/array_functions/test_array_functions_array_with_const.groovy:19]: why set to false? Is Nererids planner ok for this case?

Comment (amorynan) [reply regression-test/suites/datatype_p0/nested_types/query/array_functions/test_array_functions_array_with_const.groovy:19]: below in this groovy we have nereids query

**Review by eldenmoon — APPROVED** [review]
LGTM

File: regression-test/suites/datatype_p0/nested_types/query/array_functions/test_array_functions_array_with_const.groovy [hunk 2]
```diff
@@ -31,7 +32,16 @@ suite("test_array_functions_array_with_const", "p0") {
                 SELECT array_with_constant(-231.37104, -138); 
                 """
     } catch (Exception ex) {
-        assertTrue(ex.getMessage().contains("Array size can not be negative in function:array_with_constant"))
+        assertTrue(ex.getMessage().contains("Array size should in range(0, 1000000) in function"))
+    }
+
+    // -- {server for large array}
+    try {
```
Comment (zclllyybb) [inline regression-test/suites/datatype_p0/nested_types/query/array_functions/test_array_functions_array_with_const.groovy:39]: don't use `try-catch` to do sql failure test. it doesn't work at all. if a sql didn't fail, the test will success. it's not as expected. please use `test{ sql exception }` instead

**Review by zclllyybb — CHANGES_REQUESTED** [review]

File: be/src/vec/functions/array/function_array_with_constant.cpp [hunk 3]
```diff
@@ -91,9 +91,9 @@ class FunctionArrayWithConstant : public IFunction {
         array_sizes.reserve(input_rows_count);
         for (size_t i = 0; i < input_rows_count; ++i) {
             auto array_size = num->get_int(i);
-            if (UNLIKELY(array_size < 0)) {
-                return Status::RuntimeError("Array size can not be negative in function:" +
-                                            get_name());
+            if (UNLIKELY(array_size < 0) || UNLIKELY(array_size > max_array_size_as_field)) {
+                return Status::RuntimeError("Array size should in range(0, {}) in function: {}",
```
Comment (xiaokang) [inline be/src/vec/functions/array/function_array_with_constant.cpp:95]: add array_size actual value in error message for debug

**Review by xiaokang — APPROVED** [review]

File: be/src/vec/functions/array/function_array_with_constant.cpp (hunk 3 above)
Comment (amorynan) [reply be/src/vec/functions/array/function_array_with_constant.cpp:95]: array_size has already in error message

**Pull request merged**